source/vibe/web/rest.d

```diff
@@ -433,18 +433,22 @@ class RestInterfaceClient(I) : I
 			auto query = appender!string();
 
 			foreach (idx, p; params) {
-				static if (!(idx % 2)) {
-					// Parameter name
-					static assert(is(typeof(p) == string), "Internal error ("~__FUNCTION__~"): Parameter name is not string.");
-					query.put('&');
-					filterURLEncode(query, p);
-				} else {
-					// Parameter value
-					query.put('=');
-					static if (is(typeof(p) == Json))
-						filterURLEncode(query, p.toString());
-					else // Note: CTFE triggers compiler bug here (think we are returning Json, not string).
-						filterURLEncode(query, toRestString(serializeToJson(p)));
+				// gets value corresponding to the current parameter
+				auto value = params[(idx / 2) * 2 + 1];
+				// generates condition for inclusion of this parameter
+				enum condition = isNullable!(typeof(value)) ? q{ !value.isNull } : q{ true };
+				if (mixin(condition))
+				{
+					static if (!(idx % 2)) {
+						// Parameter name
+						static assert(is(typeof(p) == string), "Internal error ("~__FUNCTION__~"): Parameter name is not string.");
+						query.put('&');
+						filterURLEncode(query, p);
+					} else {
+						// Parameter value
+						query.put('=');
+						filterURLEncode(query, toRestString(p));
+					}
```

Comment:
>You could do it with `continue`
Either that or just putting all the logic inside the `static if (!(idx % 2))` branch, without the `else`.